backend/webdav/webdav.go

```diff
@@ -1154,19 +1163,146 @@ func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, op
 		return errors.Wrap(err, "Update mkParentDir failed")
 	}
 
-	size := src.Size()
+	if o.fs.canChunk {
+		err = o._UpdateChunked(ctx, in, src, options...)
+		if err != nil {
+			return err
+		}
+	} else {
+		size := src.Size()
+		contentType := fs.MimeType(ctx, src)
+		filePath := o.filePath()
+		extraHeaders := map[string]string{}
+		if o.fs.useOCMtime || o.fs.hasMD5 || o.fs.hasSHA1 {
+			if o.fs.useOCMtime {
+				extraHeaders["X-OC-Mtime"] = fmt.Sprintf("%d", src.ModTime(ctx).Unix())
+			}
+			// Set one upload checksum
+			// Owncloud uses one checksum only to check the upload and stores its own SHA1 and MD5
+			// Nextcloud stores the checksum you supply (SHA1 or MD5) but only stores one
+			if o.fs.hasSHA1 {
+				if sha1, _ := src.Hash(ctx, hash.SHA1); sha1 != "" {
+					extraHeaders["OC-Checksum"] = "SHA1:" + sha1
+				}
+			}
+			if o.fs.hasMD5 && extraHeaders["OC-Checksum"] == "" {
+				if md5, _ := src.Hash(ctx, hash.MD5); md5 != "" {
+					extraHeaders["OC-Checksum"] = "MD5:" + md5
+				}
+			}
+		}
+		err = o._UpdateSimple(ctx, in, filePath, size, contentType, extraHeaders,  options...)
+		if err != nil {
+			return err
+		}
+	}
+
+	// read metadata from remote
+	o.hasMetaData = false
+	return o.readMetaData(ctx)
+}
+
+
+// Standard update 
+func (o *Object) _UpdateSimple(ctx context.Context, in io.Reader, filePath string, size int64, contentType string, extraHeaders map[string]string, options ...fs.OpenOption) (err error) {
```

Comment:
preferred naming would be `updateSimple` for a private method

Reply:
✅